gnosis/safe/proxy_factory.py

```diff
@@ -191,34 +103,104 @@ def deploy_proxy_contract_with_nonce(
             master_copy, initializer, salt_nonce
         )
 
-        tx_parameters = {"from": deployer_account.address}
-        contract_address = create_proxy_fn.call(tx_parameters)
+        # Auto estimation of gas does not work. We use a little more gas just in case (gas_increment)
+        return self.deploy_contract_with_deploy_function(
+            self.ethereum_client,
+            deployer_account,
+            create_proxy_fn,
+            gas,
+            gas_price,
+            nonce,
+            gas_increment=50000,
+        )
 
-        if gas_price is not None:
-            tx_parameters["gasPrice"] = gas_price
+    @cache
+    def get_proxy_runtime_code(self, address: Optional[ChecksumAddress] = None):
+        """
+        Get runtime code for current proxy factory
+        """
+        address = address or self.address
+        return self.get_contract(address=address).functions.proxyRuntimeCode().call()
 
-        if gas is not None:
-            tx_parameters["gas"] = gas
 
-        if nonce is not None:
-            tx_parameters["nonce"] = nonce
+class ProxyFactoryV100(ProxyFactoryCommon):
+    @classmethod
+    def deploy_proxy_factory_contract(
+        cls, ethereum_client: EthereumClient, deployer_account: LocalAccount
+    ) -> EthereumTxSent:
+        """
+        Deploy proxy factory contract v1.0.0
 
-        tx = create_proxy_fn.build_transaction(tx_parameters)
-        # Auto estimation of gas does not work. We use a little more gas just in case
-        tx["gas"] = tx["gas"] + 50000
-        tx_hash = self.ethereum_client.send_unsigned_transaction(
-            tx, private_key=deployer_account.key
+        :param ethereum_client:
+        :param deployer_account: Ethereum Account
+        :return: deployed contract address
+        """
+        proxy_factory_contract = get_proxy_factory_V1_0_0_contract(ethereum_client.w3)
+        return cls.deploy_contract(
+            ethereum_client, deployer_account, proxy_factory_contract
         )
-        return EthereumTxSent(tx_hash, tx, contract_address)
 
     def get_contract(self, address: Optional[ChecksumAddress] = None):
         address = address or self.address
-        return get_proxy_factory_contract(self.ethereum_client.w3, address)
+        return get_proxy_factory_V1_0_0_contract(self.ethereum_client.w3, address)
```

Comment:
This is failing, because get_proxy_factory_... is not receiving the instance as argument. We can work around of it at the following way:
```
    @staticmethod
    def get_proxy_factory_fn(w3: Web3, address: Optional[str] = None):
        return get_proxy_factory_V1_3_0_contract(w3, address)
```